src/pages/ReimbursementAccount/VerifyIdentity/VerifyIdentity.tsx

```diff
@@ -1,51 +1,54 @@
-import PropTypes from 'prop-types';
 import React, {useCallback} from 'react';
 import {View} from 'react-native';
+import type {OnyxEntry} from 'react-native-onyx';
 import {withOnyx} from 'react-native-onyx';
 import HeaderWithBackButton from '@components/HeaderWithBackButton';
 import InteractiveStepSubHeader from '@components/InteractiveStepSubHeader';
 import ScreenWrapper from '@components/ScreenWrapper';
 import useLocalize from '@hooks/useLocalize';
 import useSubStep from '@hooks/useSubStep';
+import type {SubStepProps} from '@hooks/useSubStep/types';
 import useThemeStyles from '@hooks/useThemeStyles';
-import {reimbursementAccountPropTypes} from '@pages/ReimbursementAccount/reimbursementAccountPropTypes';
-import * as ReimbursementAccountProps from '@pages/ReimbursementAccount/reimbursementAccountPropTypes';
-import getDefaultValueForReimbursementAccountField from '@pages/ReimbursementAccount/utils/getDefaultValueForReimbursementAccountField';
+import {reimbursementAccountDefaultProps} from '@pages/ReimbursementAccount/reimbursementAccountPropTypes';
 import * as BankAccounts from '@userActions/BankAccounts';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
+import type {ReimbursementAccount} from '@src/types/onyx';
+import type {OnfidoData} from '@src/types/onyx/ReimbursementAccountDraft';
 import OnfidoInitialize from './substeps/OnfidoInitialize';
 
-const propTypes = {
+type VerifyIdentityOnyxProps = {
     /** Reimbursement account from ONYX */
-    reimbursementAccount: reimbursementAccountPropTypes,
+    reimbursementAccount: OnyxEntry<ReimbursementAccount>;
+};
 
+type VerifyIdentityProps = VerifyIdentityOnyxProps & {
     /** Goes to the previous step */
-    onBackButtonPress: PropTypes.func.isRequired,
+    onBackButtonPress: () => void;
 
     /** Exits flow and goes back to the workspace initial page */
-    onCloseButtonPress: PropTypes.func.isRequired,
-};
-
-const defaultProps = {
-    reimbursementAccount: ReimbursementAccountProps.reimbursementAccountDefaultProps,
+    onCloseButtonPress: () => void;
 };
 
-const bodyContent = [OnfidoInitialize];
+const bodyContent: Array<React.ComponentType<SubStepProps>> = [OnfidoInitialize];
 
-function VerifyIdentity({reimbursementAccount, onBackButtonPress, onCloseButtonPress}) {
+function VerifyIdentity({reimbursementAccount = reimbursementAccountDefaultProps, onBackButtonPress, onCloseButtonPress}: VerifyIdentityProps) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
 
     const submit = useCallback(
-        (onfidoData) => {
-            BankAccounts.verifyIdentityForBankAccount(getDefaultValueForReimbursementAccountField(reimbursementAccount, 'bankAccountID', 0), onfidoData);
+        (onfidoData?: OnfidoData) => {
+            if (!onfidoData) {
+                return;
+            }
```

Comment:
Have you tested before/after flow with this new condition? I just want to make sure it's okay that we skip 2 methods calls, even before it was probably called without onfido data.

Reply:
it's going to be there, this is just to satisfy the TS change to `onFinished`, as this step is unlike the others within the flow and would require a refactor in my opinion (already discussed), however this is just the migration part. (Onfido itself needs to be migrated to provide proper typings)

Reply:
It still looks strange. I've checked the code, and it looks like `onFinished` is never called with onfido data (or with any data). So your submit function will never execute methods below. Am I wrong?

Reply:
if I understand correctly, it will be called with arguments https://github.com/Expensify/App/blob/vit-tieredBankAccountFlow/src/components/Onfido/BaseOnfidoWeb.js#L82 but implicitly